src/storage/src/hummock/sstable/delete_range_aggregator.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::Ordering;
+use std::collections::{BTreeSet, BinaryHeap};
+use std::sync::Arc;
+
+use risingwave_hummock_sdk::key::{get_epoch, key_with_epoch, user_key};
+use risingwave_hummock_sdk::key_range::KeyRange;
+use risingwave_hummock_sdk::HummockEpoch;
+
+#[derive(Clone)]
+pub struct DeleteRangeTombstone {
+    start_user_key: Vec<u8>,
+    end_user_key: Vec<u8>,
```

Comment:
start_user_key is inclusive and end_user_key is exclusive? Need more documentations.

Reply:
Yes.

Reply:
> start_user_key is inclusive and end_user_key is exclusive? Need more documentations.

Yes. I have just asked.
> > Is delete range a closed interval or an open interval?
> 
> The left of it is closed while the right of it is open

Reply:
I have add comment for this